### PR TITLE
VSCode UI Tweaks and Bugfixes

### PR DESCRIPTION
**Pull request by edzkite**
* Use 𝕜𝕚𝕥𝕖 instead of ⟠ in VS Code completion item detail (fixes https://github.com/kiteco/kiteco/issues/8715)
* Restore italic font in hover text (fixes https://github.com/kiteco/kiteco/issues/8646)
* Indent nested completions (fixes https://github.com/kiteco/kiteco/issues/8672)
* Fix icon inconsistencies in completions (fixes https://github.com/kiteco/kiteco/issues/8683)

### CODE REVIEW

File: src/completion.js [hunk 1]
```diff
@@ -88,54 +119,62 @@ module.exports = class KiteCompletionProvider {
...
+          if (c.documentation_text !== "") {
+            item.documentation = buildMarkdown(
+              c.symbol.value[0].repr,
+              c.hint,
+              c.documentation_text
+            );
+          }
+          // Note: The space following the Kite icon is the unicode space U+2003
+          // instead of the normal space U+0020 because VS Code strips the detail.
+          item.detail = c.hint + " ⟠ ";
```
Comment (metalogical) [inline src/completion.js]: do we also need to change this? why is this code duplicated?

Comment (edzkite) [reply src/completion.js]: Updated. This is for the legacy completions codepath, so that's why it looks so similar.

Comment (metalogical) [reply src/completion.js]: would it make sense to factor out values such as this into top-level constants?

Comment (edzkite) [reply src/completion.js]: Yeah, absolutely. Will do before merging.

**Pull request merged**